test/e2e/internalencryption/internalencryption_test.go

```diff
@@ -0,0 +1,175 @@
+	url := resources.Route.Status.URL.URL()
+	if _, err := pkgTest.CheckEndpointState(
+		context.Background(),
+		clients.KubeClient,
+		t.Logf,
+		url,
+		spoof.MatchesAllOf(spoof.IsStatusOK, spoof.MatchesBody(test.HelloWorldText)),
+		"HelloWorldText",
+		test.ServingFlags.ResolvableDomain,
+		test.AddRootCAtoTransport(context.Background(), t.Logf, clients, test.ServingFlags.HTTPS),
+	); err != nil {
+		t.Fatalf("The endpoint %s for Route %s didn't serve the expected text %q: %v", url, names.Route, test.HelloWorldText, err)
+	}
+
+	// Check on the logs for the activator
+	pods, err := clients.KubeClient.CoreV1().Pods(system.Namespace()).List(context.TODO(), v1.ListOptions{
+		LabelSelector: "app=activator",
+	})
+	if err != nil {
+		t.Fatalf("Failed to get pods: %v", err)
+	}
+	if len(pods.Items) == 0 {
+		t.Fatalf("No pods detected for activator: %v", err)
+	}
+	activatorPod := pods.Items[0]
+
+	req := clients.KubeClient.CoreV1().Pods(activatorPod.Namespace).GetLogs(activatorPod.Name, &corev1.PodLogOptions{})
+	activatorTLSCount, err := scanPodLogs(req, matchTLSLog)
+
+	if err != nil {
+		t.Fatalf("Failed checking activator logs: %s", err)
+	} else if activatorTLSCount == 0 {
+		t.Fatal("TLS not used on requests to activator")
+	}
+
+	// Check on the logs for the queue-proxy
+	pods, err = clients.KubeClient.CoreV1().Pods("serving-tests").List(context.TODO(), v1.ListOptions{
+		LabelSelector: fmt.Sprintf("serving.knative.dev/configuration=%s", names.Config),
+	})
+	if err != nil {
+		t.Fatalf("Failed to get pods: %v", err)
+	}
+	if len(pods.Items) == 0 {
+		t.Fatalf("No pods detected for test app: %v", err)
+	}
+	helloWorldPod := pods.Items[0]
+	req = clients.KubeClient.CoreV1().Pods(helloWorldPod.Namespace).GetLogs(helloWorldPod.Name, &corev1.PodLogOptions{Container: "queue-proxy"})
+	queueTLSCount, err := scanPodLogs(req, matchTLSLog)
+
+	if err != nil {
+		t.Fatalf("Failed checking queue-proxy logs: %s", err)
+	} else if queueTLSCount == 0 {
+		t.Fatal("TLS not used on requests to queue-proxy")
+	}
+}
+
+func scanPodLogs(req *rest.Request, matcher func(string) bool) (matchCount int, err error) {
+
+	podLogs, err := req.Stream(context.Background())
+	if err != nil {
+		err = fmt.Errorf("Failed to stream activator logs: %w", err)
+		return
+	}
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	podLogs.Close()
+	if err != nil {
+		err = fmt.Errorf("Failed to read activator logs from buffer: %w", err)
+		return
+	}
+
+	scanner := bufio.NewScanner(buf)
+	for scanner.Scan() {
+		if matcher(scanner.Text()) {
+			matchCount++
+		}
+	}
+
+	if err = scanner.Err(); err != nil {
+		err = fmt.Errorf("Failed scanning activator logs: %w", err)
+		return
+	}
+
+	return
+}
+
+func matchTLSLog(line string) bool {
+	if strings.Contains(line, "TLS") {
+		if strings.Contains(line, "TLS: <nil>") {
+			return false
+		} else if strings.Contains(line, "TLS: [") {
```

Comment:
```suggestion
		} else if strings.Contains(line, "TLS: {") {
```
It must be a cause of current failure :sunglasses:

Reply:
oh good catch, I think I was using a different access log format in my local cluster